### PR TITLE
Boxes3D archetype & port of old box api

### PR DESCRIPTION
**Pull request by Wumpf**
### What

* fixes *most* of #2786 

Introduces two new components:
* `HalfSizes3D`
* `Rotation3D`

Only remaining things on Box2D/3D are:
* python extension methods (similar to rust/cpp ones)
* gaps in C++ support on other archetypes caused by Box2D missing until now

image

*Commit by commit review possible, massaged the history accordingly.*

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested [demo.rerun.io](https://demo.rerun.io/pr/3339) (if applicable)

- [PR Build Summary](https://build.rerun.io/pr/3339)
- [Docs preview](https://rerun.io/preview/136868d768cf48b277db0b4adeabacf9fb9f15b1/docs) 
- [Examples preview](https://rerun.io/preview/136868d768cf48b277db0b4adeabacf9fb9f15b1/examples) 
- [Recent benchmark results](https://ref.rerun.io/dev/bench/)
- [Wasm size tracking](https://ref.rerun.io/dev/sizes/)

### CODE REVIEW

File: crates/re_types/definitions/rerun/archetypes/boxes3d.fbs [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+  /// Optional text labels for the boxes.
+  labels: [rerun.components.Text] ("attr.rerun.component_optional", nullable, order: 3000);
+
+  /// Optional `ClassId`s for the boxes.
+  ///
+  /// The class ID provides colors and labels if not specified explicitly.
+  class_ids: [rerun.components.ClassId] ("attr.rerun.component_optional", nullable, order: 3200);
+
+  /// Unique identifiers for each individual boxes in the batch.
+  instance_keys: [rerun.components.InstanceKey] ("attr.rerun.component_optional", nullable, order: 3400);
```
Comment (teh-cmc) [inline crates/re_types/definitions/rerun/archetypes/boxes3d.fbs:86]: a couple of copy-paste issues with `order` attributes

**Review by teh-cmc — APPROVED** [review]

File: crates/re_types/definitions/rerun/archetypes/boxes3d.fbs (hunk 1 above)
Comment (Wumpf) [reply crates/re_types/definitions/rerun/archetypes/boxes3d.fbs:86]: huh kind suprised I don't get shouted at for having the same `order` twice!

Comment (teh-cmc) [reply crates/re_types/definitions/rerun/archetypes/boxes3d.fbs]: #2680 is why :grimacing:

Comment (Wumpf) [reply crates/re_types/definitions/rerun/archetypes/boxes3d.fbs]: good reason :D

**Pull request merged**